### PR TITLE
fix: ignore client sent option UUID and generate new UUID on column update

### PR DESCRIPTION
**Pull request by samin-z**
Issue: #2897 

**Behaviour:**

Upon column update, any UUID sent in the request payload is ignored (to prevent client manipulation and to preserve the server managed identity). existing options keep their stored UUID and new options get a newly generated one.

### 🏁 Checklist
 
- [x] ⛑️ Tests (unit and/or integration) are included or not needed
- [ ] 🔙 Backport requests are created or not needed: `/backport to stableX.X`
- [ ] 📅 Milestone is set
- [x] 🌸 PR title is meaningful (if it should be in the changelog: is it meaningful to users?)

## 🤖 AI (if applicable)

- [x] The content of this PR was partly or fully generated using AI

### CODE REVIEW

File: lib/Model/SelectionOption.php [hunk 1]
```diff
@@ -49,7 +47,29 @@ public function label(): string {
...
+	 */
+	public function ensureUuid(?string $existingUuid = null): void {
+		if ($this->uuid() !== null) {
+			return;
+		}
+		if ($existingUuid !== null && $existingUuid !== '') {
+			$this->setUuid($existingUuid);
+			return;
+		}
+		$this->setUuid(Uuid::v7()->toRfc4122());
```
Comment (blizzz) [inline lib/Model/SelectionOption.php:69]: Not sure having it here. It should be exclusively generated when creating an option first time.

**Review by blizzz — COMMENTED** [review]
Brief look, left a comment. But a UUID should never be regenerated, it must remain stable once assigned upon creation.

File: lib/Model/SelectionOption.php (hunk 1 above)
Comment (samin-z) [reply lib/Model/SelectionOption.php:69]: thanks for the review, i refactored it.

File: lib/Model/SelectionOptions.php [hunk 2]
```diff
@@ -95,6 +95,34 @@ public static function createFromInputJsonString(?string $data, null|bool|int|st
...
+		if ($this->selectionOptions === null) {
+			return;
+		}
+
+		$existingByKey = [];
+		if ($existing !== null) {
+			foreach ($existing as $existingOption) {
+				$existingByKey[$existingOption->key()] = $existingOption;
+			}
+		}
```
Comment (enjeck) [inline lib/Model/SelectionOptions.php:111]: Do we need to worry about protecting agains t `TypeError` if stored options are `null` ? 
```suggestion
		if ($existing !== null && $existing->jsonSerialize() !== null) {
			foreach ($existing as $existingOption) {
				$existingByKey[$existingOption->key()] = $existingOption;
			}
		}
```

File: lib/Model/SelectionOptions.php [hunk 3]
```diff
@@ -95,6 +95,34 @@ public static function createFromInputJsonString(?string $data, null|bool|int|st
 		return self::createFromInputArray($inputArray, $default, $allowPassingUuid);
 	}
 
+	/**
+	 * UUIDs are never regenerated or taken from client input
```
Comment (enjeck) [inline lib/Model/SelectionOptions.php:99]: is this comment still true? `Column::fromDto` parses with `allowPassingUuid = true`?

File: tests/unit/Model/SelectionOptionUuidTest.php
Comment (enjeck) [inline tests/unit/Model/SelectionOptionUuidTest.php:1]: maybe add a test for the existing options being null. The potential issue I mentioned earlier

File: lib/Model/SelectionOption.php [hunk 4]
```diff
@@ -49,7 +47,35 @@ public function label(): string {
...
+			return;
+		}
+		$this->setUuid($uuid);
+	}
+
+	/**
+	 * Generate a UUID when creating an option for the first time, existing ones are not replaced
+	 */
+	public function generateUuid(): void {
+		if ($this->uuid() !== null) {
```
Comment (enjeck) [inline lib/Model/SelectionOption.php]: Is it necessary to check if `uuid() !== null` ? Wondering if the guard in`setUuid` is enough already

File: lib/Model/SelectionOptions.php (hunk 3 above)
Comment (blizzz) [reply lib/Model/SelectionOptions.php:99]: Create from Tables Structure should set them. And upcoming structure update as well.

File: lib/Model/SelectionOptions.php (hunk 2 above)
Comment (blizzz) [reply lib/Model/SelectionOptions.php:111]: When `$existing` is not null, it is `SelectionOption` and then it must have a key. So the `jsonSerialize()` check is not necessary.

File: lib/Model/SelectionOption.php (hunk 4 above)
Comment (blizzz) [reply lib/Model/SelectionOption.php]: ack, seems duplicated

**Review by blizzz — APPROVED** [review]
Tested successfully on top of table-scheme-import

Added a commit to re-run migration for the selection options

File: lib/Model/SelectionOption.php (hunk 4 above)
Comment (samin-z) [reply lib/Model/SelectionOption.php]: the duplicated check is already removed, setUuid() is no-ops if a UUID is set.
